das/daser.go

```diff
@@ -15,28 +15,152 @@ import (
 	"github.com/celestiaorg/celestia-node/share"
 )
 
-var log = logging.Logger("das")
+var (
+	log = logging.Logger("das")
 
-// TODO: parameters needs performance testing on real network to define optimal values
-const (
-	//  samplingRange is the maximum amount of headers processed in one job.
-	samplingRange = 100
+	ErrInvalidOptionValue = func(optionName string, value string) error {
+		errorMsg := fmt.Sprintf("das/daser: invalid option value: %s, cannot be %s", optionName, value)
+		return errors.New(errorMsg)
+	}
+)
+
+// optionSetter:
+//
+//	this is expected to be a closure that encloses the operation of parameter setting along side with
+//	the value to be set.
+//	example: both d and a are in the "global scope" relatively to the closure, thus no parameter passing is required.
+//	   func() {
+//
+// .       d.params.option = a
+//
+//	}
+type optionSetter func()
+type Option func(*DASer) error
+type OptionValidator func(value any, optionSetter optionSetter) error
+
+type parameters struct {
+	samplingRange     uint
+	concurrencyLimit  uint
+	bgStoreInterval   time.Duration
+	priorityQueueSize uint
+	genesisHeight     uint
+}
```

Comment:
There seems to be a lot of new code for options. Let's move it into the separate file